### PR TITLE
Jetpack: adds logic for when to display Activity Log filter + small fixes

### PR DESCRIPTION
**Pull request by leandroalonso**
Part of #15193

* Displays the Activity Log filters only if the site has a paid plan OR has the backup capability
* Remove dots from some strings
* Fixes an event property
* Changes the number of items per page for Backups to 100

### To test

Review the strings and the event change. :)

Then:

1. Run the app
2. Tap Activity Log for a free site
3. Check that the filters do not appear
4. Open Activity Log in a paid or Jetpack site with backup
5. Check that the filters do appear
6. Tap Backup
7. Check that more items are loaded initially

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Models/JetpackSiteRef.swift [hunk 1]
```diff
@@ -31,4 +37,8 @@ struct JetpackSiteRef: Hashable, Codable {
         return lhs.siteID == rhs.siteID
             && lhs.username == rhs.username
```
Comment (momo-ozawa) [inline WordPress/Classes/Models/JetpackSiteRef.swift:38]: WDYT of adding checks for `hasBackup` and `hasPaidPlan` for `==(lhs: rhs:)` just to be safe?

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift [hunk 2]
```diff
@@ -18,7 +18,7 @@ class ActivityListViewModel: Observable {
     private let noResultsTexts: ActivityListConfiguration
     private var storeReceipt: Receipt?
 
-    private let count = 20
+    var numberOfItemsPerPage = 20
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: Better name, thanks! 👍

**Review by momo-ozawa — COMMENTED** [review]
Works as described 👍

File: WordPress/Classes/Models/JetpackSiteRef.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/Models/JetpackSiteRef.swift:38]: Sure, good catch!

Done in 03b314dd274da2303f0c40d8379f7e6c418ab4bf

**Review by momo-ozawa — APPROVED** [review]

**Pull request merged**